core/src/main/groovy/org/docToolchain/http/BasicRestClient.groovy

```diff
@@ -3,15 +3,27 @@ package org.docToolchain.http
 import org.apache.hc.client5.http.impl.classic.CloseableHttpClient
 import org.apache.hc.client5.http.impl.classic.HttpClientBuilder
 import org.apache.hc.core5.http.ClassicHttpRequest
+import org.apache.hc.core5.http.Header
+import org.apache.hc.core5.http.HttpHeaders
 import org.apache.hc.core5.http.HttpHost
 import org.apache.hc.core5.http.io.HttpClientResponseHandler
+import org.apache.hc.core5.net.URIBuilder
 
 abstract class BasicRestClient {
 
     protected HttpClientBuilder httpClientBuilder
 
     BasicRestClient() {
         this.httpClientBuilder = HttpClientBuilder.create()
+        httpClientBuilder.addRequestInterceptorFirst { request, entityDetails, context ->
+            request.setHeader(HttpHeaders.USER_AGENT, "docToolchain_v${getClass().getPackage().getImplementationVersion()}")
+            Header hostHeader = request.getHeader(HttpHeaders.HOST)
+            if (hostHeader == null) {
```

Comment:
@PacoVK Are you sure that this will fix the discussed issue with the port in the `Host` header? Maybe we should in all cases set the header manually? Not sure.

Reply:
Unless you do not set it explicitly, i am sure at least the docs say for `addRequestInterceptorFirst` "Adds this protocol interceptor to the head of the protocol processing list." ([see](https://hc.apache.org/httpcomponents-client-5.2.x/current/httpclient5/apidocs/org/apache/hc/client5/http/impl/classic/HttpClientBuilder.html)) whilst the standard `RequestTargetHost` Interceptor later in the chain first checks, whether a `Host` header has already been set. ([see](https://github.com/apache/httpcomponents-core/blob/master/httpcore5/src/main/java/org/apache/hc/core5/http/protocol/RequestTargetHost.java)) :)

Reply:
Although i.am not sure if the check is necessary and we would have ever the need to set custom `Host` header, this at least has then the same behaviour as the standard interceptor